server/sonar-web/src/main/js/app/integration/vsts/components/Configuration.tsx

```diff
@@ -0,0 +1,189 @@
+      return this.props.widgetHelpers.WidgetConfigurationSave.Invalid();
+    }
+    return this.props.widgetHelpers.WidgetConfigurationSave.Valid({
+      data: JSON.stringify(this.state.settings)
+    });
+  };
+
+  fetchOrganizations = () => {
+    this.setState({ loading: true });
+    getOrganizations({ member: true }).then(
+      ({ organizations }) => {
+        if (this.mounted) {
+          this.setState({ organizations, loading: false });
+        }
+      },
+      () => {
+        // TODO Handle error
+      }
+    );
+  };
+
+  fetchProjects = (organization?: string) => {
+    this.setState({ loading: true });
+    searchProjects({ organization, ps: 100 }).then(
+      ({ components }) => {
+        if (this.mounted) {
+          this.setState({
+            projects: components.map(c => ({ label: c.name, value: c.key })),
+            loading: false
+          });
+        }
+      },
+      () => {
+        this.setState({
+          projects: [],
+          loading: false
+        });
+      }
+    );
+  };
+
+  handleProjectSearch = (query: string) => {
+    return searchProjects({
+      organization: this.state.organizations,
+      ps: 50,
+      filter: `query="${query}"`
+    }).then(({ components }) => components.map(c => ({ label: c.name, value: c.key })));
+  };
+
+  handleProjectSelect = ({ value }: { value: string }) => {
+    this.setState(
+      ({ settings }) => ({ settings: { ...settings, project: value } }),
+      this.notifyChange
+    );
+  };
+
+  handleProjectChange = (
+    event: React.ChangeEvent<HTMLSelectElement> | React.FocusEvent<HTMLSelectElement>
+  ) => {
+    const { value } = event.currentTarget;
+    this.setState(
+      ({ settings }) => ({ settings: { ...settings, project: value } }),
+      this.notifyChange
+    );
+  };
+
+  notifyChange = ({ settings, widgetConfigurationContext } = this.state) => {
+    const { widgetHelpers } = this.props;
+    if (widgetConfigurationContext && widgetConfigurationContext.notify) {
+      const eventName = widgetHelpers.WidgetEvent.ConfigurationChange;
+      const eventArgs = widgetHelpers.WidgetEvent.Args({ data: JSON.stringify(settings) });
+      widgetConfigurationContext.notify(eventName, eventArgs);
+    }
+  };
+
+  render() {
+    const { projects, loading, settings } = this.state;
+    if (loading) {
+      return (
+        <div className="vsts-loading">
+          <i className="spinner global-loading-spinner" />
+        </div>
+      );
+    }
+    return (
+      <div className="widget-configuration">
+        <div className="dropdown" id="project">
+          <label>SonarCloud project</label>
+          <div className="wrapper">
+            <select
+              onBlur={this.handleProjectChange}
+              onChange={this.handleProjectChange}
+              value={settings.project}>
+              <option disabled={true} hidden={true} value="">
+                Select a project...
```

Comment:
Not really since it's on SonarCloud. But more seriously, I don't want to download the full bundle into the widget... Maybe if we switch to a plugin we could localize it.